src/content/blog/astro-4150.mdx

```diff
@@ -0,0 +1,184 @@
+
+In Astro 4.15, you can now specify a `timeout` property that will trigger the island to load if it has not already done so in that time. This is useful to add priority to a component that might be competing with other idle components.
+
+```astro
+{/* Many many components */}
+{Array.from({ length: 3000 }).map((_, i) => (<Counter id={`client-idle-${i}`} count={i} client:idle>
+	<h1>Hello, client:idle! ${i}</h1>
+</Counter>))}
+
+{/* Component loads after a maximum wait of 500ms */}
+<Counter id="client-idle-timeout" {...someProps} client:idle={{ timeout: 500 }}>
+	<h1>Hello, client:idle={'{{timeout: 500}}'}!</h1>
+</Counter>
+```
+
+Thanks to [ph1p](https://github.com/ph1p) for this contribution.
+
+## Swap helpers for ViewTransitions
+
+Astro's `<ViewTransitions />` router works by turning your MPA (multi-page app) into a SPA (single-page app), while attempting to preserve the DX of a MPA. It does this by fetching pages on navigation and swapping the content. In some cases, you might want more control over how the swap works, which can be done by overriding the swap in the [`astro:before-swap`](https://docs.astro.build/en/guides/view-transitions/#astrobefore-swap) event.
+
+In Astro 4.15, the built-in swap functions are now exported from `astro:transitions/client`. These helpers allow you to use parts of what Astro does by default, with your own custom logic in between.
+
+```astro
+<script>
+import { swapFunctions } from 'astro:transitions/client';
+
+document.addEventListener('astro:before-swap', (e) => {
+  e.swap = () => swapMainOnly(e.newDocument);
+});
+
+function swapMainOnly(doc: Document) {
+  swapFunctions.deselectScripts(doc);
+  swapFunctions.swapRootAttributes(doc);
+  swapFunctions.swapHeadElements(doc);
+  const restoreFocusFunction = swapFunctions.saveFocus();
+  const newMain = doc.querySelector('main');
+  const oldMain = document.querySelector('main');
+  if (newMain && oldMain) {
+    swapFunctions.swapBodyElement(newMain, oldMain);
+  } else {
+    swapFunctions.swapBodyElement(doc.body, document.body);
+  }
+  restoreFocusFunction();
+}
+<script>
+```
+
+Thanks to [martrapp](https://github.com/martrapp) for this contribution.
+
+## Bug Fixes and Special Thanks
+
+As we do, Astro 4.16 includes more bug fixes and smaller improvements that couldn't make it into this post! Check out the full [release notes](https://github.com/withastro/astro/blob/refs/heads/main/packages/astro/CHANGELOG.md#4130) to learn more.
+
+Thanks to [Sarah Rainsberger (@sarah11918)](https://github.com/sarah11918), [Yan (@yanthomasdev)](https://github.com/yanthomasdev), [Bjorn Lu (@bluwy)](https://github.com/bluwy) and everyone else who contributed to this release.
```

Comment:
Just curious how we are deciding?

- Ben wrote the actions part?
- Chris *relentlessly* helped edit the Actions docs
- Ema had two features this release
-  We've had some pretty helpful docs reviews throughout this! (Lorenzo, Armand) and one of the Turso people helped with definitive terminology/spelling for the libSQL stuff.

Reply:
I didn't edit this, but typically I just take whoever I see in the PRs